openedx_tagging/core/tagging/import_export/dsl.py

```diff
@@ -0,0 +1,182 @@
+"""
+Model and functions to create a plan/execution with DSL actions.
+"""
+from typing import List, Optional
+
+from django.utils.translation import gettext_lazy as _
+
+from ..models import Taxonomy
+from .actions import (
+    DeleteTag,
+    ImportAction,
+    UpdateParentTag,
+    WithoutChanges,
+    available_actions,
+)
+
+
+class TagDSL:
```

Comment:
You can use [dataclasses](https://docs.python.org/3/library/dataclasses.html) to reduce boilerplate code for these utility classes. Or if that's not enough, there's also [attrs](https://www.attrs.org/en/stable/examples.html) -- edx-platform imports it (but doesn't use it much), so I think it would be ok to add to the requirements here.

Reply:
Just writing to confirm that `attrs` is fine to add as a dependency. It's the recommended lib to use for this kind of thing in [OEP-49](https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0049-django-app-patterns.html#data-py). Feel free to use its built-in validation mechanisms as well.

Reply:
Updated here: https://github.com/openedx/openedx-learning/pull/64/commits/332db7b50d54f1a133f112609daa3c8438664027